src/tasks/FilterFeedTask.ts

```diff
@@ -0,0 +1,83 @@
+import { MentionRegistry } from '../mentions/MentionRegistry';
+import { FilterFeedConfig } from '../BotConfig';
+import { Client, TextChannel } from 'discord.js';
+import * as log4js from 'log4js';
+import Task from './Task';
+import JiraClient = require( 'jira-connector' );
+
+export default class FilterFeedTask extends Task {
+	public static logger = log4js.getLogger( 'FilterFeed' );
+
+	private client: Client;
+	private jira: JiraClient;
+	private channel: string;
+	private jql: string;
+	private title: string;
+
+	private knownTickets: Set<string>;
+
+	constructor( { channel, jql, title }: FilterFeedConfig, client: Client ) {
+		super();
+
+		this.channel = channel;
+		this.jql = jql;
+		this.title = title;
+		this.client = client;
+
+		this.jira = new JiraClient( {
+			host: 'bugs.mojang.com',
+			strictSSL: true,
+		} );
+	}
+
+	public async run(): Promise<void> {
+		let upcomingTickets: string[];
+
+		try {
+			const searchResults = await this.jira.search.search( {
```

Comment:
Attention: `this` doesn't refer to `FilterFeedTask` here, since you run this method using `setInterval`! JavaScript/TypeScript is tricky in this regard.

```
[2020-01-12T14:18:49.575] [ERROR] FilterFeed - TypeError: Cannot read property 'search' of undefined
    at Timeout.<anonymous> (...\mojira-discord-bot\src\tasks\FilterFeedTask.ts:37:42)
    at Generator.next (<anonymous>)
    at ...\mojira-discord-bot\src\tasks\FilterFeedTask.ts:8:71
    at new Promise (<anonymous>)
    at __awaiter (...\mojira-discord-bot\src\tasks\FilterFeedTask.ts:4:12)
    at Timeout.run [as _onTimeout] (...\mojira-discord-bot\src\tasks\FilterFeedTask.ts:30:16)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
```